modules/channel-web/src/views/lite/core/socket.tsx

```diff
@@ -51,23 +47,33 @@ export default class BpSocket {
   }
 
   /** Waits until the VISITOR ID and VISITOR SOCKET ID is set  */
-  public waitForUserId(): Promise<void> {
-    return new Promise((resolve, reject) => {
-      const interval = setInterval(() => {
-        if (this.isString(window.__BP_VISITOR_ID) && this.isString(window.__BP_VISITOR_SOCKET_ID)) {
-          clearInterval(interval)
+  public async waitForUserId(): Promise<void> {
+    await waitForUserId()
```

Comment:
this can throw after 30s what happens then ? wenbchat crashes ?

Reply:
It would mean that the webchat wasn't able to properly initialized itself in 30sec and will simply throw an error that is displayed in the console.